examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -7,6 +7,7 @@
 #include <numeric>
 #include <string>
 #include <unistd.h>
+#include <memory>
```

Comment:
personally I like to have headers ordered alphabetically, it makes the code more tidy

Reply:
Let's make a deal:
I can order headers alphabetically if we agree to order includes from specific to general.

https://blog.knatten.org/2010/07/01/the-order-of-include-directives-matter/

Reply:
Interesting :-)

I do not agree with his conclusions "This all works out nicely until someone else wants to include geometry.hpp without including <vector> first, and get something like geometry.hpp:1: error: ‘vector’ is not a member of ‘std’. You have now forced all the users of your geometry library to include <vector> before including geometry.hpp."

What I would conclude is: I have forced the person who wrote geometry.hpp to fix their _BUG_ and include vector inside geometry.hpp ;-)

There are several different opinions, again. Google have their own, different from mine, different from yours: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

Anyway, on this one (unlike the #ifdefs!) I have not such strong opinions. It does not matter. 

Maybe let's see what Stefan or others think. As long as we have a convention and stick with that, fine by me. (I am sure that each convention will include alphabetical ordering within each block, even if different conventions order blocks in different orders!)

Reply:
> What I would conclude is: I have forced the person who wrote geometry.hpp to fix their BUG and include vector inside geometry.hpp ;-)

That's exactly the point. He phrases this wrongly, but including in the order
```
#include "MyHeader.h"

#include <other stuff>
```
uncovers bugs.

Reply:
Boh, I am not convinced but I see your point. Pick your convention on includes - but not on #ifdefs, lets keep memcpy there.

I suggest the google convention, IIUC it fits your requirements, and is a standard. Else choose...
```
In dir/foo.cc or dir/foo_test.cc, whose main purpose is to implement or test the stuff in dir2/foo2.h, 
order your includes as follows:

    dir2/foo2.h.
    A blank line
    C system headers (more precisely: headers in angle brackets with the .h extension), e.g., <unistd.h>, <stdlib.h>.
    A blank line
    C++ standard library headers (without file extension), e.g., <algorithm>, <cstddef>.
    A blank line
    Other libraries' .h files.
    Your project's .h files.

Separate each non-empty group with one blank line.

With the preferred ordering, if the related header dir2/foo2.h omits any necessary includes, 
the build of dir/foo.cc or dir/foo_test.cc will break.
...
Within each section the includes should be ordered alphabetically. 
```